### PR TITLE
Replace search menu with search bar

### PR DESCRIPTION
**Pull request by ashiagr**
## Description
This replaces the search menu with a search bar on the podcasts screen.

(Part of up next in tab bar and profile tab changes)

## Testing Instructions
1. Start with no subscribed podcasts
2. Open podcasts tab 
3. ✅ Notice that you see a search bar and the search menu is removed
4. Subscribe to a few podcasts (at least 15 in count) 
5. ✅ Notice that search bar is still shown
6. Scroll to the end of the list and then slightly scroll to the top
7. ✅ Notice that search bar is hidden on scrolling towards bottom of the list and reappears on scroll to the top (you do not have to scroll to the top of the list to see the search bar)
8. Create a folder and add a few podcasts to it
9. Tap on folder
10. ✅ Notice that search bar is not shown
11. Tap on search bar
12. ✅ Notice that full-screen search screen is shown
13. ✅ Notice that you still see `search_shown` track event on tapping search bar
14. ✅  Notice that adding search bar has not caused any regression to pull-to-refresh or podcast rearrange behavior

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/1405144/aab3036f-3022-48ab-9599-c4a90a267d22

Blank screen

Dark theme | Light theme
----|----
 | 

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md - To be added
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

File: modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/components/SearchBar.kt [hunk 1]
```diff
@@ -155,7 +169,7 @@ private fun SearchBarDarkPreview() {
 @Composable
 private fun SearchBarPreview() {
     Column(modifier = Modifier.padding(8.dp)) {
-        SearchBar("Material", onTextChanged = {}, modifier = Modifier.padding(bottom = 8.dp))
+        SearchBar("Materiaxdzsdasd asdasdas sdadsa asdadsa asdasdas asdsada asdadsl", onTextChanged = {}, modifier = Modifier.padding(bottom = 8.dp))
```
Comment (MiSikora) [inline modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/components/SearchBar.kt:172]: I laughed 😆

**Review by MiSikora — APPROVED** [review]
Really nice improvement.

I think that the search bar could use a little bit more padding under it but I'm not 100% sure.

Untitled

File: modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/components/SearchBar.kt (hunk 1 above)
Comment (mebarbosa) [reply modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/components/SearchBar.kt:172]: Who has never done this? 😆

Comment (ashiagr) [reply modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/components/SearchBar.kt]: Oops! 🙈 reverted in https://github.com/Automattic/pocket-casts-android/pull/2077/commits/0812d506d87c1ff417c1b8b22b9fddc6c6fd77f3

**Comment by ashiagr** [comment]
> I think that the search bar could use a little bit more bottom padding but I'm not 100% sure.

I updated padding in https://github.com/Automattic/pocket-casts-android/pull/2077/commits/334aa22c84c8900d99b07569a06737977a7bf8d2:

**Pull request merged**